### PR TITLE
{AKS} support image cleaner in public preview

### PR DESCRIPTION
**Pull request by jiashun0011**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 1]
```diff
@@ -886,6 +886,111 @@ def get_azure_keyvault_kms_key_vault_resource_id(self) -> Union[str, None]:
...
+        enable_image_cleaner = self.raw_param.get("enable_image_cleaner")
+
+        return enable_image_cleaner
+
+    def get_enable_image_cleaner(self) -> bool:
+        """Obtain the value of enable_image_cleaner.
+
+        :return: bool
+        """
+        return self._get_enable_image_cleaner()
```
[the remark this answers is not in the document]
Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:904]: Synced offline, may use a validator to perform mutually exclusive validation.

File: src/aks-preview/azext_aks_preview/_validators.py [hunk 2]
```diff
@@ -607,6 +607,10 @@ def validate_azure_keyvault_kms_key_vault_resource_id(namespace):
     if not is_valid_resource_id(key_vault_resource_id):
         raise InvalidArgumentValueError("--azure-keyvault-kms-key-vault-resource-id is not a valid Azure resource ID.")
 
+def validate_image_cleaner_interval_hours(namespace):
+    image_cleaner_interval_hours = namespace.image_cleaner_interval_hours
+    if image_cleaner_interval_hours and not isinstance(image_cleaner_interval_hours, int):
+        raise InvalidArgumentValueError("--image-cleaner-interval-hours is not a valid integer.")
```
[the remark this answers is not in the document]
Comment (jiashun0011) [reply src/aks-preview/azext_aks_preview/_validators.py:615]: Removed this extra integer check.

[the remark this answers is not in the document]
Comment (jiashun0011) [reply src/aks-preview/azext_aks_preview/_validators.py:615]: > I could see you've declared `image_cleaner_interval_hours` should be `int` in `_param.py`, I guess this validator is not necessary?

Removed this extra integer check.

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jiashun0011) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:904]: > Synced offline, may use a validator to perform mutually exclusive validation.

Already put this mutually exclusive check in validator.

[the remark this answers is not in the document]
Comment (jiashun0011) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py]: > I suppose you could merge these 2 functions.

Combined these internal and external function.

**Review by FumingZhang — APPROVED** [review]
LGTM

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 3]
```diff
@@ -3926,6 +3926,133 @@ def test_aks_update_with_workload_identity(self, resource_group, resource_group_
             self.check('securityProfile.workloadIdentity.enabled', False),
         ])
 
+    @live_only()
```
Comment (zhoxing-ms) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:3929]: May I ask why those new tests are `@live_only()`

Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:3929]: Ah, the backend service is not ready, so the tests (also the new options) could not work. The RP release is in progress and should be available soon.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**